### PR TITLE
Add documentation for exposing Kiali and Grafana with oauth2_proxy

### PR DESCRIPTION
**Pull request by chrkl**
**Description**

Changes proposed in this pull request:

- Add documentation for exposing Kiali and Grafana with oauth2_proxy

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/11228

### CODE REVIEW

File: docs/operation-guides/config-07-kiali-grafana.md [hunk 1]
```diff
@@ -44,3 +44,247 @@ Kyma does not expose Kiali and Grafana by default. However you can still access
...
+            number: 3000
+          host: oauth2-kiali
+  ```
+  </details>
+</div>
+  <details>
+  <summary>
+  Grafana
+  </summary>
+  1. Chose a domain for the exposed service under the Kyma base domain. E.g., `grafana.kyma.example.com` if your Kyma cluster is reachable under `kyma.example.com`.
```
Comment (NHingerl) [inline docs/operation-guides/config-07-kiali-grafana.md:175]:
```suggestion
  1. Chose a domain for the exposed service under the Kyma base domain. For example, if your Kyma cluster is reachable under `kyma.example.com`, use `grafana.kyma.example.com` .
```

File: docs/operation-guides/config-07-kiali-grafana.md [hunk 2]
```diff
@@ -44,3 +44,247 @@ Kyma does not expose Kiali and Grafana by default. However you can still access
...
+
+  2. Create a new Github application under https://github.com/settings/apps. Set the callback URL to `https://grafana.kyma.example.com/oauth2/callback`. Ensure at least read-only access to email addresses for the Github application. Copy the client id and secret.
```
Comment (NHingerl) [inline docs/operation-guides/config-07-kiali-grafana.md:177]:
```suggestion
  2. Create a new Github application under https://github.com/settings/apps. Set the callback URL to `https://grafana.kyma.example.com/oauth2/callback`. Ensure at least read-only access to email addresses for the Github application. Copy the client ID and secret.
```

File: docs/operation-guides/config-07-kiali-grafana.md [hunk 3]
```diff
@@ -44,3 +44,247 @@ Kyma does not expose Kiali and Grafana by default. However you can still access
...
+
+  3. Create a Kubernetes Secret for the client id and secret:
```
Comment (NHingerl) [inline docs/operation-guides/config-07-kiali-grafana.md:179]:
```suggestion
  3. Create a Kubernetes Secret for the client ID and secret:
```

File: docs/operation-guides/config-07-kiali-grafana.md [hunk 4]
```diff
@@ -44,3 +44,247 @@ Kyma does not expose Kiali and Grafana by default. However you can still access
...
+      route:
+      - destination:
+          port:
+            number: 3000
+          host: oauth2-grafana
+  ```
+  </details>
+</div>
+
+The `oauth2_proxy` supports a wide range of other well known authentication services or OpenID Connect for custom solutions. See the [list of supported providers](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/oauth_provider) to find instructions for other authentication services.
```
Comment (NHingerl) [inline docs/operation-guides/config-07-kiali-grafana.md:290]:
```suggestion
>**NOTE:** The `oauth2_proxy` supports a wide range of other well-known authentication services or OpenID Connect for custom solutions. See the [list of supported providers](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/oauth_provider) to find instructions for other authentication services.
```

I wonder whether this information would be better above the step description, as it explains why we call our instructions an "example".  Wouldn't readers want to know this first; before reading all the steps?

**Review by NHingerl — CHANGES_REQUESTED** [review]

File: docs/operation-guides/config-07-kiali-grafana.md (hunk 4 above)
Comment (chrkl) [reply docs/operation-guides/config-07-kiali-grafana.md:290]: Good suggestion. Will move it.

**Comment by chrkl** [comment]
Closing in favour of https://github.com/kyma-project/kyma/pull/11403

**Pull request closed without merging**